services/src/main/java/org/keycloak/services/resources/account/AccountFormService.java

```diff
@@ -1090,5 +1141,12 @@ private void csrfCheck(final MultivaluedMap<String, String> formData) {
         }
     }
 
+    private void checkRealmAndClientRoles() {
```

Comment:
Why is this a separate method? Why is it checking both client and realm roles? There shouldn't be a realm level delete account role, only a client level associated with account client.